identity-service/src/awsSNS.js

```diff
@@ -138,43 +141,77 @@ async function drainPublishedAnnouncements () {
 // we want it to continue
 async function drainMessageObject (bufferObj) {
   const { userId } = bufferObj
-  const { deviceType, awsARN, deviceToken } = bufferObj.device
   const { message, title, playSound } = bufferObj.notificationParams
 
-  try {
-    let formattedMessage = null
-    if (deviceType === 'ios') {
-      // TODO - add badge count here
-      formattedMessage = _formatIOSMessage(message, awsARN, playSound, title)
+  logger.debug(`Retrieving badge count for user ${userId}`)
+  const incrementBadgeQuery = await models.PushNotificationBadgeCounts.increment('iosBadgeCount', { where: { userId } })
+  // Parse the updated value returned from increment
+  // TODO: Is this actually the right access pattern?
+  const newBadgeCount = incrementBadgeQuery[0][0][0].iosBadgeCount
+  /*
+  const userBadgeQuery = await models.PushNotificationBadgeCounts.findOrCreate({
+    where: {
+      userId
     }
-
-    if (formattedMessage) {
-      await publishPromisified(formattedMessage)
+  })
+  const userBadgeObj = userBadgeQuery[0]
+  const newBadgeCount = userBadgeObj.dataValues.iosBadgeCount + 1
+  const updateResp = await models.PushNotificationBadgeCounts.update(
+    {
+      iosBadgeCount: newBadgeCount
+    },
+    {
+      where: {
+        userId
+      }
     }
-  } catch (e) {
-    if (e && e.code && (e.code === 'EndpointDisabled' || e.code === 'InvalidParameter')) {
-      try {
-        // this notification is not deliverable to this device
-        // remove from deviceTokens table and de-register from AWS
-        const tokenObj = await models.NotificationDeviceToken.findOne({
-          where: {
-            deviceToken,
-            userId
-          }
-        })
+  )
+  logger.debug(`Updated badge count - ${updateResp}`)
+  */
+  logger.debug(`New badge count ${newBadgeCount}`)
+
+  const devices = await models.NotificationDeviceToken.findAll({ where: { userId } })
+  // If no devices found, short-circuit
+  if (devices.length === 0) return
+  // Dispatch to all devices
+  await Promise.all(devices.map(async (device) => {
```

Comment:
have you tested the try catch inside this promise.all to make sure it doesn't early exit on invalid device tokens? seems like this should work without changing publishPromisified

Reply:
im not clear how this changes publishPromisifed at all? i can definitely test this though